src/gd_avif.c

```diff
@@ -356,6 +356,15 @@ BGD_DECLARE(gdImagePtr) gdImageCreateFromAvifCtx (gdIOCtx *ctx)
 
 	decoder = avifDecoderCreate();
 
+	// Check if libavif version is >= 0.9.1.
+	// If so, allow the PixelInformationProperty ('pixi') to be missing in AV1 image
+	// items. libheif v1.11.0 or older does not add the 'pixi' item property to
+	// AV1 image items. (This issue has been corrected in libheif v1.12.0.)
+
+#if AVIF_VERSION_MAJOR > 0 || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR > 9) || (AVIF_VERSION_MAJOR == 0 && AVIF_VERSION_MINOR == 9 && AVIF_VERSION_PATCH >= 1)
```

Comment:
these type of compares are so hard to read, and so easy to get wrong

isn't `AVIF_VERSION` provided specifically to make checks easier ?

```
#if AVIF_VERSION >= 90100
...
#endif
```

Reply:
No; see https://github.com/php/php-src/pull/7253#discussion_r671560824 for alternatives (none of which appear really better to me).

Reply:
assuming you mean this part:
> This assumes the way AVIF_VERSION is defined in <avif/avif.h> will never change:

i don't see the problem here.  it wouldn't be a publicly exported symbol if people weren't expected to rely on it, and the header file explicitly says:
> This should allow for downstream projects to do greater-than preprocessor checks on AVIF_VERSION.

so assuming the format of the define is fine.

if upstream disagrees, then they should fix/clarify the header, and probably provide a macro like `AVIF_MAKE_VERSION` so people could pass in a tuple like `AVIF_VERSION >= AVIF_MAKE_VERSION(0, 9, 1)`.

Reply:
A tuple of this sort would certainly be convenient!

Paging @wantehchang to get his opinion.

Reply:
@vapier, ah, I misread the comment. Doing `#if AVIF_VERSION >= 90100` is fine with me.

Reply:
I think the comment text quoted above actually is an explication of the start of the comment:

`AVIF_VERSION_DEVEL should always be 0 for official releases / version tags,
and non-zero during development of the next release.`

Nonetheless, since `AVIF_VERSION_DEVEL` does give this far more compact and legible way to determine the precise version, and it seems reasonable to expect it will continue to exist in this form, I'm personally quite happy to rely on it. I've made the change.